### ISSUE TITLE
Show which tfsec rule prompted the error

### ISSUE THREAD

**Issue by jeffgran** [reviewdog/action-tfsec#16]
It would be nice to show the tfsec rule identifier in the comments that reviewdog leaves. This is important for false positives / acceptable risk, because it tells the developer which rule to ignore with a comment on that line. 

```
$ tfsec

Problem 1

  [AWS002][ERROR] Resource 'module.blah.aws_s3_bucket.default' does not have logging enabled.
```

The `AWS002` is the critical part that's missing.

**Comment by Vlaaaaaaad** [comment]
**Tl;dr**: There's not much we can do in this action to fix this 😞 The only way to fix the problem _here_ would be to write and use a custom converter for the `tfsec` output, but that could be very brittle.
  
  
  --- 

Let's go into some detail! 

On the one hand, `tfsec` can output its findings in a couple of formats: 
- `default` which is the nice formatting you see in the terminal
-  `json` which dumps a JSON string
- `csv` which dumps a CSV sting
- **`checkstyle`** which dumps a checkstyle-formatted XML
- `junit`which dumps a junit-formatted XML


On the other hand, `reviewdog` accepts a couple of different output from linters, as per [the docs](https://github.com/reviewdog/reviewdog#input-format): 
- `errorformat` which comes from the Vim world
- `diff` which is just a diff output (the same one you see when doing a `git diff`, if you're not familiar with `diff`)
- **`checkstyle`** which is a loosely-defined XML format that comes from the Java world
- Reviewdog Diagnostic Format which we'll get into later
 
To make `tfsec` talk directly to `reviewdog` we use the **`checkstyle`** format. It's the only one supported by both tools!

If we go in the `testdata` folder and run `tfsec --format checkstyle` we get the following output:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle>
  <file name="/workspace/action-tfsec/testdata/example.tf">
    <error 
      rule="AZU003" 
      line="43" 
      column="0" 
      severity="ERROR" 
      message="Resource &#39;azurerm_managed_disk.example&#39; defines an unencrypted managed disk." 
      link="See https://tfsec.dev/docs/azure/AZU003/ for more information.">
    </error>

    <error 
      rule="AWS006" 
      line="24" 
      column="0" 
      severity="WARNING" 
      message="Resource &#39;aws_security_group_rule.example_rule&#39; defines a fully open ingress security group rule." 
      link="See https://tfsec.dev/docs/aws/AWS006/ for more information.">
    </error>
  </file>
<checkstyle>
```

We see that each `tfsec` rule violation has a couple of values: `rule`, `line`&`column`, `severity`, `message`, and `link`.

On the other hand, `reviewdog` [reads only some values](https://github.com/reviewdog/reviewdog/blob/3acb0871ee10438b934efbb266050ebddd93c748/parser/checkstyle.go#L72-L78):  `line`&`column`, `severity`, `message`, and `source`.


🤦 The `checkstyle` format does not have strict definitions, so we have a mismatch: `rule` and `link` are used on the `tfsec` side and `source` is expected on the `reviewdog` side.
This is the exact reason why @haya14busa went and created the Reviewdog Diagnostic Format which you can read about [here](https://github.com/reviewdog/reviewdog/tree/master/proto/rdf). We need a strictly defined format to get all the tools to talk to each other! 

Going back to the original issue, we don't have many options:
- have `tfsec` output the rule name in the message. Maybe even have it be a hyperlink to the tfsec rule docs? We can create an issue in [tfsec/tfsec](https://github.com/tfsec/tfsec) to see if they want to implement this breaking change
- have `reviewdog` parse the tfsec-special `rule` and `link` values. I don't know how widely they are used, and I don't think reviewdog should have tool-specific fields
- somehow make `tfsec` and `reviewdog` talk nicely together. This means one of the following:
  * have `tfsec` output JSON, parse and re-format that JSON into Reviewdog Diagnostic Format, and send that to `reviewdog`
  * have `tfsec` output `checkstyle`, do a bunch of manipulation of that XML, and send it to `reviewdog`
  
Neither option is pretty or easy to do. I would accept a PR that makes `tfsec` and `reviewdog` talk nicely together, but I fear that might be brittle.

Does anybody have any other ideas for how to fix this?

**Comment by haya14busa** [comment]
FYI: action-hadolint uses JSON output and convert it to rdjson format by using jq.
https://github.com/reviewdog/action-hadolint/blob/d75ed2297c693a99f137250f0035ae53cadedc26/script.sh#L36-L37

**Comment by Vlaaaaaaad** [comment]
> FYI: action-hadolint uses JSON output and convert it to rdjson format by using jq.
> https://github.com/reviewdog/action-hadolint/blob/d75ed2297c693a99f137250f0035ae53cadedc26/script.sh#L36-L37

OOoh, I like that! Yes, I'd definitely accept a PR implementing this! I'd do it myself, but I don't know when I'll get to it.

**Comment by jplanckeel** [comment]
Can you check this ? i have created a PR for add url tfsec in reviewdog https://github.com/reviewdog/action-tfsec/pull/26
